http_client.rst

```diff
@@ -1378,12 +1378,25 @@ This allows using them where native PHP streams are needed::
     // later on if you need to, you can access the response from the stream
     $response = stream_get_meta_data($streamResource)['wrapper_data']->getResponse();
 
-Testing HTTP Clients and Responses
-----------------------------------
+Testing
+-------
 
 This component includes the ``MockHttpClient`` and ``MockResponse`` classes to
-use them in tests that need an HTTP client which doesn't make actual HTTP
-requests.
+use them in tests that shouldn't make actual HTTP requests. Such tests can be
+useful, as they are faster and consistent due to not being dependant on
+external service the request would be sent to. In result there is no need to
+worry about the service being online or the request changing some state,
+for example deleting a resource.
```

Comment:
```suggestion
useful, as they will run faster and produce consistent results, since they're not dependant on an external service. By not making actual HTTP requests there
is no need to worry about the service being online or the request changing state, for example deleting a resource.
```